### PR TITLE
Add phantomjs module

### PR DESCRIPTION
**Pull request by ppp0**
To run QUnit tests (https://github.com/cargomedia/cm/pull/2140), we need to have [phamtomjs][phantomjs] installed in dev and CI env.

Some inspiration: https://forge.puppet.com/3fs/phantomjs

[phantomjs]: http://phantomjs.org

### CODE REVIEW

**Comment by ppp0** [comment]
Here our installer (just `require 'phantomjs'`)

This installs a statically-linked binary. According to http://phantomjs.org/download.html#linux-64-bit I've also required to package `fontconfig` but without being able to verify its correct installation / need

@njam please review

**Comment by njam** [comment]
- Why is "build" needed?
- Defining the "unp" package will conflict with puppet-cargomedia. How about using original tooling to do the job?
 - Alternatively we could use this stdlib function [ensure_package](https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/lib/puppet/parser/functions/ensure_packages.rb) they use in the linked module.

**Comment by ppp0** [comment]
* oops.. you're right - a leftover from some gist I took an initial inspiration of
* agree `tar xjf` does the job
* I put it on, what are the upsides exactly versus using `package` which does an existence check, too?

@njam please re-review

**Comment by njam** [comment]
Having two `package{'foo':}` in the catalog will result in a conflict. This function seems to check if the same resource is already defined (with the same parameters).

One more thing: Will `install.sh` work when an (older) version is already installed? Maybe just `rm -rf /usr/local/share/phantomjs` first?

**Comment by njam** [comment]
Ah, and for `ensure_packages` we should pass `provider => 'apt'`.

**Comment by ppp0** [comment]
* true, let's purge the directory before

**Comment by ppp0** [comment]
@njam please review

File: modules/phantomjs/Puppetfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+mod 'puppetlabs/stdlib',
+  :git => 'git://github.com/puppetlabs/puppetlabs-stdlib.git'
+
+mod 'cargomedia/helper',
+  :git => 'git@github.com:cargomedia/puppet-packages.git',
+  :path => 'modules/helper
```
Comment (njam) [inline modules/phantomjs/Puppetfile]: missing quote

**Comment by njam** [comment]
small syntax error in Puppetfile, rest looks good

**Comment by ppp0** [comment]
trcky one! @njam please review

**Comment by njam** [comment]
Spec:
```
Notice: /Stage[main]/Phantomjs/Helper::Script[install phantomjs]/Exec[exec install phantomjs]/returns: bzip2: (stdin) is not a bzip2 file.
```

**Comment by ppp0** [comment]
retest this please

**Pull request merged**